partner_identification/models/res_partner.py

```diff
@@ -94,50 +98,56 @@ def _inverse_identification(self, field_name, category_code):
-                    'category_id': category.id,
-                    'name': name,
-                })
+        # For the moment do nothing with empty value.
+        if not field_name or not field_value:
+            return
+        # Check, and if needed autocreate, category:
+        category_model = self.pool['res.partner.id_category']
+        category_ids = category_model.search(
+            cr, uid, [('code', '=', category_code)], context=context)
+        if not category_ids:
+            category_id = category_model.create(
+                cr, uid, {
+                    'code': category_code,
+                    'name': category_code},
+                context=context)
+        else:
+            category_id = category_ids[0]
+        id_model = self.pool['res.partner.id_number']
+        for record in self.browse(cr, uid, [ids], context=context):
+            # Search al records with the right category.
+            id_number_ids = id_model.search(
+                cr, uid, [
+                    ('partner_id', '=', record.id),
+                    ('category_id', '=', category_id)],
+                context=context)
+            if len(id_number_ids) > 1:
+                # Guard against writing wrong records.
+                raise orm.except_orm(
+                    _('Error'),
+                    _('This %s has multiple IDs of this type (%s), so a write'
+                      ' via the %s field is not possible.\n'
+                      'In order to fix this, please use the IDs tab.') % (
+                          record._name, category_code, field_name))
+            if len(id_number_ids) < 1:
+                id_model.create(
+                    cr, uid, {
+                        'partner_id': record.id,
+                        'category_id': category_id,
+                        'name': field_value},
+                    context=context)
+                return
             # There was an identification record singleton found.
-            elif record_len == 1:
-                value = record[field_name]
-                if value:
-                    id_number.name = value
-                else:
-                    id_number.active = False
-            # Guard against writing wrong records.
-            else:
-                raise ValidationError(_(
-                    'This %s has multiple IDs of this type (%s), so a write '
-                    'via the %s field is not possible. In order to fix this, '
-                    'please use the IDs tab.',
-                ) % (
-                    record._name, category_code, field_name,
-                ))
-
-    def _search_identification(self, category_code, operator, value):
-        """ Search method for an identification field.
+            id_model.write(
+                cr, uid, id_number_ids, {
+                    'partner_id': record.id,
+                    'category_id': category_id,
+                    'name': field_value},
```

Comment:
just writing `name` seems sufficient to me, those ids come back from searching for the other two fields

Reply:
True